### DISCUSSION TITLE
New Certifcates are all signed by an unknown authority.

### DISCUSSION THREAD

**Discussion by NoobSpielt** [smallstep/certificates#1249] [Q&A] [answered]
Hey!

I am quite new with step ca so probably I just did some stupid error.

I installed Step-Ca with that Documentation: [https://smallstep.com/docs/step-ca/getting-started](https://smallstep.com/docs/step-ca/getting-started)

With that I created a root and a intermedia ca cert.

Now when I create a client cert via this command: `step ca certificate example.com --san example.com --san 10.0.50.10 eaxmple.com.crt example.com.key` It seems fine, it creates the cert and we are all happy :P.

But now when I want to acces the the website with https it says the cert is not trusted. With that I tried using `step certificate inspect https://example.com`. I got following error: `failed to connect: x509: certificate signed by unknown authority`

But when i try `step certificate inspect https://example.com --roots .step/certs/intermediate_ca.crt` everything seems to be working great,

So why does my machine only trust the root ca and not the intermedia and if I need to trust the intermedia also why isnt it automatacly installed on the machine with: `step ca bootstrap` ?

**Comment by maraino** [comment, marked as answer]
You need to install your root certificate in your truststore. To do this you can run:

```
step certificate install ~/.step/certs/root_ca.crt
```

See the flags using `step certificate install --help` for the install options.

**Reply by NoobSpielt** [reply]
Hey, thanks for the fast response. 
Sorry i forget to give the information that I already used `step certificate install $(step path)/certs/root_ca.crt` and it installed the root ca into my machine.

**Reply by maraino** [reply]
For the inspect command, you should pass the roots unless the root is installed in the system. `step certificate` commands do not trust `step ca`-related roots. You can try with:

```
step certificate inspect --roots $(step path)/certs/root_ca.crt https://example.com
```
Does this work?

AFAIK Chrome and Safari use the system truststore, but others like firefox use their own truststore, we have a specific flag for it `--firefox`. But if it's something different there will settings to configure it.

Different Linux distributions might be using different ways to install the root, we support multiple cases, but perhaps not all of them, here you can see some of the ones we try https://github.com/smallstep/truststore/blob/bf50add3b063e7fad56f92be62df9b591e89edcf/truststore_linux.go#L30-L46

And finally other CLI utilities like `curl` might be using OpenSSL truststore or your system one, it depends on how they are built. If it's using OpenSSL, you might want to manually add your root_ca.crt.

**Reply by NoobSpielt** [reply]
hmm. I only tried curl for that, but `step certificate inspect --roots $(step path)/certs/root_ca.crt https://example.com` didnt work either. Only with the intermedia cert.

**Reply by maraino** [reply]
> Only with the intermedia cert.

Oh, then the problem is that the certificate served by your server should include the intermediate cert too.

When you do `step ca certificate example.com example.crt example.key`, the dot crt file includes both server certificate an intermediate, and that's what it should be use. 

Check for example this:
```
step certificate inspect --format pem --bundle --short https://smallstep.com
```

-- it shows multiple intermediates

**Reply by NoobSpielt** [reply]
ahh so the crt for example.com should include the intermedia and the normal cert I see thanks for that.

But is there any way to skip that without installing the intermedia cert on any machine? I have 2 machines were it is not possible for me to include the intermedia.

**Reply by maraino** [reply]
If you cannot add the intermediate the only solution is to install also the intermediate into your truststore. But those machines are not really following the X.509 standard. They won't work if, for example, you want to open those to the internet using a public PKI.

It's also possible to generate leaf certificates from the root, so the intermediate is not necessary, but you cannot really do that with `step-ca`. You should use `step certificate create` for that.

**Reply by NoobSpielt** [reply]
Thank u very much for all that good information! It helped me really much ❤️
